src/metal.cs

```diff
@@ -378,6 +437,32 @@ public partial interface MTLBlitCommandEncoder : MTLCommandEncoder {
 
 		[Abstract, Export ("copyFromBuffer:sourceOffset:toBuffer:destinationOffset:size:")]
 		void CopyFromBuffer (IMTLBuffer sourceBuffer, nuint sourceOffset, IMTLBuffer destinationBuffer, nuint destinationOffset, nuint size);
+		
+		[iOS (10,0), TV (10,0), NoWatch, NoMac]
+		[Abstract]
+		[Export ("updateFence:")]
+		void UpdateFence (IMTLFence fence);
+
+		[iOS (10,0), TV (10,0), NoWatch, NoMac]
+		[Abstract]
+		[Export ("waitForFence:")]
+		void WaitForFence (IMTLFence fence);
+	}
+	
+	public interface IMTLFence {}
+
+	[iOS (10,0), TV (10,0), NoWatch, NoMac]
+	[Protocol, Model]
```

Comment:
it's not a delegate so let's not create a `[Model]` as it's not clear a concrete implementation would be useful (but could be confusing and missed in API)